src/flint/types/fmpz_mod.pyx

```diff
@@ -62,6 +70,30 @@ cdef class fmpz_mod_ctx:
         fmpz_set(n.val, (<fmpz_t>self.val.n))
         return n
 
+    cdef any_as_fmpz_mod(self, obj):
+        # If `obj` is an `fmpz_mod`, just check moduli
+        # match
+        # TODO: we could allow conversion from one modulus to another?
+        if typecheck(obj, fmpz_mod):
+            if self != (<fmpz_mod>obj).ctx:
+                raise ValueError("moduli must match")
+            return obj
```

Comment:
I'm not sure exactly how the `!=` is implemented at the C level here but if there was something to ensure that all contexts for a given modulus are unique in memory then this could be like a pointer comparison.